src/shell/output.c

```diff
@@ -325,7 +379,9 @@ struct shell_output *shell_output_create (flux_shell_t *shell)
             errno = ENOMEM;
             goto error;
         }
-        if (shell_output_header_append (shell, out->output) < 0)
+        if (!(out->pending_commits = zlist_new ()))
+            goto error;
+        if (shell_output_header (out) < 0)
```

Comment:
You mentioned you wanted failure to commit the header to be fatal, but it looks like if the commit RPC returns an error, it is non-fatal in `shell_output_commit_completion()`?

Reply:
ahh good catch.  I was primarily concerned with fatal error on the request side, had forgotten the response side.

Reply:
hmmm, on such an error would it be better to run `flux_shell_killall()` rather than just exit?  B/c all tasks have started.

Reply:
I think we need a `shell_fatal_error ()` or similar call that attempts to clean up before a fatal error when tasks have started, instead of issuing the killall at the site of the error. Or perhaps with some thinking we could come up with a cleaner way to handle fatal errors with proper cleanup -- haven't thought about it too much at this point.

Reply:
yeah, I was just starting to think that.  Perhaps for now, `log_err_exit()` is fine and we can figure out the proper / best / cleanest way to deal with this in another issue.